docs/tracing.md

```diff
@@ -0,0 +1,237 @@
+---
+jupytext:
+  formats: md:myst
+  text_representation:
+    extension: .md
+    format_name: myst
+    format_version: 0.13
+    jupytext_version: 1.16.4
+kernelspec:
+  display_name: Python 3
+  language: python
+  name: python3
+---
+
+(tracing-tutorial)=
+# Tracing
+
+To use `jax.jit` effectively, it is useful to understand how it works. Let's put
+a few `print()` statements within a JIT-compiled function and then call the function:
+
+```{code-cell}
+from jax import jit
+import jax.numpy as jnp
+import numpy as np
+
+@jit
+def f(x, y):
+  print("Running f():")
+  print(f"  x = {x}")
+  print(f"  y = {y}")
+  result = jnp.dot(x + 1, y + 1)
+  print(f"  result = {result}")
+  return result
+
+x = np.random.randn(3, 4)
+y = np.random.randn(4)
+f(x, y)
+```
+
+Notice that the print statements execute, but rather than printing the data we
+passed to the function, though, it prints *tracer* objects that stand-in for
+them (something like `Traced<ShapedArray(float32[])>`).
+
+These tracer objects are what `jax.jit` uses to extract the sequence of
+operations specified by the function. Basic tracers are stand-ins that encode
+the **shape** and **dtype** of the arrays, but are agnostic to the values. This
+recorded sequence of computations can then be efficiently applied within XLA to
+new inputs with the same shape and dtype, without having to re-execute the
+Python code.
+
+When we call the compiled function again on matching inputs, no re-compilation
+is required and nothing is printed because the result is computed in compiled
+XLA rather than in Python:
+
+```{code-cell}
+x2 = np.random.randn(3, 4)
+y2 = np.random.randn(4)
+f(x2, y2)
+```
+
+The extracted sequence of operations is encoded in a JAX expression, or
+[*jaxpr*](key-concepts-jaxprs) for short. You can view the jaxpr using the
+`jax.make_jaxpr` transformation:
+
+```{code-cell}
+from jax import make_jaxpr
+
+def f(x, y):
+  return jnp.dot(x + 1, y + 1)
+
+make_jaxpr(f)(x, y)
+```
+
+Note one consequence of this: because JIT compilation is done *without*
+information on the content of the array, control flow statements in the function
+cannot depend on traced values. For example, this fails:
```

Comment:
Could you add "...cannot depend on traced values (see [Control flow and logical operators with JIT](https://docs.jax.dev/en/latest/control-flow.html) for more details)"?